internal/project/repository.go

```diff
@@ -15,6 +15,8 @@ var (
 
 type Repository interface {
 	GetAllProjects(ctx context.Context) (*[]Project, error)
+	GetProjectByID(ctx context.Context, id int) (*Project, error)
+	GetProjectVersions(ctx context.Context, projectID int) (*[]Version, error)
```

Comment:
Any reason for returning a pointer to slice?
The usual reason we return pointer to object is for the zero-value (in case of errors) being `nil`
Since the zero-value of a slice is already `nil` we usually just return the slice.
Unless of cause there are performance considerations, but I doubt that here.